pkg/api/rest/create.go

```diff
@@ -57,7 +57,14 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx api.Context, obj runtime.Obje
 			return errors.NewBadRequest("the namespace of the provided object does not match the namespace sent on the request")
 		}
 	} else {
-		objectMeta.Namespace = api.NamespaceNone
+		if ns, ok := api.NamespaceFrom(ctx); ok && ns != api.NamespaceNone {
+			return errors.NewBadRequest("namespace is not allowed on this type")
```

Comment:
> is this a situation that could have happened with legacy url format when ?namespace= could have been provided on any resource even if not namespaced? wondering if this is something you actually saw happen with v1beta3 style urls

Definitely happens with v1beta1 style URLs. Also can happen when a registry is called directly.